### ISSUE TITLE
re-connect seems to crash discovery server

### ISSUE THREAD

**Issue by serapath** [RangerMauve/discovery-swarm-web#6]
1. I start the discovery server:
`> discovery-swarm-web`

2. I start budo:
`budo demo.js:bundle.js -open --live --ssl`
and also tried `budo demo.js:bundle.js -open --live` to not serve my site over https.

3. My website uses `dat-js` like:
`new require('dat-js')({ discovery: 'http://localhost:3472' })`

This seems to work, but as soon as i edit my project and save the changes, `budo` live reloads it, which causes `dat-js` to connect again to the discovery server, which makes it crash with:
```
events.js:177
      throw er; // Unhandled 'error' event
      ^

Error: read ECONNRESET
    at TCP.onStreamRead (internal/stream_base_commons.js:183:27)
Emitted 'error' event at:
    at emitErrorNT (C:\Users\bommel\Desktop\TEMP code\wallet-filesystem\node_modules\readable-stream\lib\internal\streams\destroy.js:68:8)
    at processTicksAndRejections (internal/process/task_queues.js:84:9) {
  errno: 'ECONNRESET',
  code: 'ECONNRESET',
  syscall: 'read',
  type: 'error',
  target: WebSocket { ... }
```

**Comment by RangerMauve** [comment]
Thanks for the bug report! I'll be working on it today. :)

**Comment by RangerMauve** [comment]
What version of dat-js are you using?

**Comment by RangerMauve** [comment]
Thanks for the bug report! This is fixed in v1.4.1 😁
